editor/doc_tools.cpp

```diff
@@ -464,60 +464,69 @@ void DocTools::generate(bool p_basic_types) {
 			c.constants.push_back(constant);
 		}
 
-		//theme stuff
-
+		// Theme items.
 		{
 			List<StringName> l;
-			Theme::get_default()->get_constant_list(cname, &l);
-			for (const StringName &E : l) {
-				DocData::PropertyDoc pd;
-				pd.name = E;
-				pd.type = "int";
-				pd.default_value = itos(Theme::get_default()->get_constant(E, cname));
-				c.theme_properties.push_back(pd);
-			}
 
-			l.clear();
 			Theme::get_default()->get_color_list(cname, &l);
 			for (const StringName &E : l) {
-				DocData::PropertyDoc pd;
-				pd.name = E;
-				pd.type = "Color";
-				pd.default_value = Variant(Theme::get_default()->get_color(E, cname)).get_construct_string();
-				c.theme_properties.push_back(pd);
+				DocData::ThemeItemDoc tid;
+				tid.name = E;
+				tid.type = "Color";
+				tid.data_type = "color";
+				tid.default_value = Variant(Theme::get_default()->get_color(E, cname)).get_construct_string();
+				c.theme_properties.push_back(tid);
 			}
 
 			l.clear();
-			Theme::get_default()->get_icon_list(cname, &l);
+			Theme::get_default()->get_constant_list(cname, &l);
 			for (const StringName &E : l) {
-				DocData::PropertyDoc pd;
-				pd.name = E;
-				pd.type = "Texture2D";
-				c.theme_properties.push_back(pd);
+				DocData::ThemeItemDoc tid;
+				tid.name = E;
+				tid.type = "int";
+				tid.data_type = "constant";
+				tid.default_value = itos(Theme::get_default()->get_constant(E, cname));
+				c.theme_properties.push_back(tid);
 			}
+
 			l.clear();
 			Theme::get_default()->get_font_list(cname, &l);
 			for (const StringName &E : l) {
-				DocData::PropertyDoc pd;
-				pd.name = E;
-				pd.type = "Font";
-				c.theme_properties.push_back(pd);
+				DocData::ThemeItemDoc tid;
+				tid.name = E;
+				tid.type = "Font";
+				tid.data_type = "font";
+				c.theme_properties.push_back(tid);
 			}
+
 			l.clear();
 			Theme::get_default()->get_font_size_list(cname, &l);
 			for (const StringName &E : l) {
-				DocData::PropertyDoc pd;
-				pd.name = E;
-				pd.type = "int";
-				c.theme_properties.push_back(pd);
+				DocData::ThemeItemDoc tid;
+				tid.name = E;
+				tid.type = "int";
+				tid.data_type = "font_size";
+				c.theme_properties.push_back(tid);
+			}
+
+			l.clear();
+			Theme::get_default()->get_icon_list(cname, &l);
+			for (const StringName &E : l) {
+				DocData::ThemeItemDoc tid;
+				tid.name = E;
+				tid.type = "Texture2D";
+				tid.data_type = "icon";
+				c.theme_properties.push_back(tid);
 			}
+
 			l.clear();
 			Theme::get_default()->get_stylebox_list(cname, &l);
 			for (const StringName &E : l) {
-				DocData::PropertyDoc pd;
-				pd.name = E;
-				pd.type = "StyleBox";
-				c.theme_properties.push_back(pd);
+				DocData::ThemeItemDoc tid;
+				tid.name = E;
+				tid.type = "StyleBox";
+				tid.data_type = "style";
```

Comment:
This is the only one that doesn't match its `get_<data_type>_list`, so should maybe be `stylebox`?

Reply:
Alright, then keeping `style` seems good.